Resolve: Update to latest npm dependencies (19-10-2021)

gromxyz opened the pull request:
Verify commands:
- [x] `yarn prebuild`
- [x] `yarn generate:types` 
- [x] `yarn build-storybook`
- [x] `yarn storybook`

Verify functionality:
- [x] Swap
- [x] Add/Remove liquidity
- [x] Tx history
- [x] Upgrade `ETH.RUNE` and `BNB.RUNE`
- [x] Send keystore/ledger
- [x] Swap UTXO

Not updated:

- `@storybook/react`  - mentioned in #1579
- `ts-loader` -  mentioned in #1411
- `cosmos-client` - waiting on xchainjs/xchainjs-lib#436 in order to upgrade both
- `rxjs` - a lot of changes in the code required to support newest version
- Stick with `usb-detection-4.10.0` (see https://github.com/thorchain/asgardex-electron/pull/1864#discussion_r731995104)


Closes #1863
Closes #1862

gromxyz committed:
Update dependencies

gromxyz committed:
Update ledger related dependencies

veado commented on package.json:
```diff
@@ -99,80 +99,80 @@
     "cross-env": "^7.0.3",
-    "electron": "^15.1.1",
-    "electron-builder": "^22.11.7",
+    "electron": "^15.2.0",
```
Does `Electron@15.2.0` fix #1862?

veado reviewed (approved):
LGTM

veado reviewed (changes requested):
## Issue
Latest changes do not work on Linux, `yarn dev` failed:
```bin
[dev:electron] A JavaScript error occurred in the main process
[dev:electron] Uncaught Exception:
[dev:electron] Error: The module 'node_modules/usb-detection/build/Release/detection.node'
[dev:electron] was compiled against a different Node.js version using
[dev:electron] NODE_MODULE_VERSION 89. This version of Node.js requires
[dev:electron] NODE_MODULE_VERSION 98. Please try re-compiling or re-installing
```

## Following steps did not work:

- Install everything from scratch `rm yarn.lock` && `rm -rf node_modules` && yarn`
- Re-build native modules using `./node_modules/.bin/electron-rebuild` (as described in README of `node-usb-detection`, chapter [Install for Electron](https://github.com/MadLittleMods/node-usb-detection#install-for-electron))
- Running different Node versions: v15 + v16
- ~~Downgrade "@ledgerhq/hw-transport": "^6.2.0", "@ledgerhq/hw-transport-node-hid-singleton": "^6.2.0" ([same versions](https://github.com/thorchain/ledger-thorchain-js/blob/main/package.json#L25) in `ledger-thorchain-js`)~~ see UPDATE below

## UPDATE:

- Downgrade to previous version of "@ledgerhq/hw-transport-node-hid-singleton" did not work, because it still changes to latest `usb-detection@0.13.0` in `yarn.lock`
- Commit 567eb33 (Downgrade `@ledgerhq/hw-transport-node-hid-singleton` AND stick `usb-detection@4.10.0`) finally fixes all this issue - check https://github.com/thorchain/asgardex-electron/pull/1864#discussion_r731995104

veado committed:
Stick @ledgerhq/hw-transport-node-hid-singleton@6.7.0

+ update yarn.lock to stick with usb-detection@4.10.0 by reverting
changes in `resolved` url

All needed to avoid issues of running usb-detection behind the scenes -
see
https://github.com/thorchain/asgardex-electron/pull/1864#pullrequestreview-783149970
for more details

veado commented on yarn.lock:
```diff
@@ -19086,15 +19112,15 @@ url@^0.11.0:
+usb-detection@^4.10.0:
+  version "4.10.0"
+  resolved "https://registry.yarnpkg.com/usb-detection/-/usb-detection-4.10.0.tgz#0f8a3b8965a5e4e7fbee1667971ca97e455ed11f"
+  integrity sha512-YUzVWXwfSviE2pInXCKYXhR5heY9GUzlWsdZYxb/Br1Xela6P31A0KDHm7XW0Wsku1HwrokZx+/OD8cZSPHR3w==
```
Sticking to sources of `usb-detection-4.10.0` are important here.
